src/main/java/spark/utils/ResourceUtils.java

```diff
@@ -0,0 +1,353 @@
+
+    /**
+     * URL protocol for a file in the file system: "file"
+     */
+    public static final String URL_PROTOCOL_FILE = "file";
+
+    /**
+     * URL protocol for an entry from a jar file: "jar"
+     */
+    public static final String URL_PROTOCOL_JAR = "jar";
+
+    /**
+     * URL protocol for an entry from a zip file: "zip"
+     */
+    public static final String URL_PROTOCOL_ZIP = "zip";
+
+    /**
+     * URL protocol for an entry from a JBoss jar file: "vfszip"
+     */
+    public static final String URL_PROTOCOL_VFSZIP = "vfszip";
+
+    /**
+     * URL protocol for an entry from a WebSphere jar file: "wsjar"
+     */
+    public static final String URL_PROTOCOL_WSJAR = "wsjar";
+
+    /**
+     * Separator between JAR URL and file path within the JAR
+     */
+    public static final String JAR_URL_SEPARATOR = "!/";
+
+
+    /**
+     * Return whether the given resource location is a URL:
+     * either a special "classpath" pseudo URL or a standard URL.
+     *
+     * @param resourceLocation the location String to check
+     * @return whether the location qualifies as a URL
+     * @see #CLASSPATH_URL_PREFIX
+     * @see java.net.URL
+     */
+    public static boolean isUrl(String resourceLocation) {
+        if (resourceLocation == null) {
+            return false;
+        }
+        if (resourceLocation.startsWith(CLASSPATH_URL_PREFIX)) {
+            return true;
+        }
+        try {
+            new URL(resourceLocation);
+            return true;
+        } catch (MalformedURLException ex) {
+            return false;
+        }
+    }
+
+    /**
+     * Resolve the given resource location to a {@code java.net.URL}.
+     * <p>Does not check whether the URL actually exists; simply returns
+     * the URL that the given location would correspond to.
+     *
+     * @param resourceLocation the resource location to resolve: either a
+     *                         "classpath:" pseudo URL, a "file:" URL, or a plain file path
+     * @return a corresponding URL object
+     * @throws FileNotFoundException if the resource cannot be resolved to a URL
+     */
+    public static URL getURL(String resourceLocation) throws FileNotFoundException {
+        Assert.notNull(resourceLocation, "Resource location must not be null");
+        if (resourceLocation.startsWith(CLASSPATH_URL_PREFIX)) {
+            String path = resourceLocation.substring(CLASSPATH_URL_PREFIX.length());
+            URL url = ClassUtils.getDefaultClassLoader().getResource(path);
+            if (url == null) {
+                String description = "class path resource [" + path + "]";
+                throw new FileNotFoundException(
+                        description + " cannot be resolved to URL because it does not exist");
+            }
+            return url;
+        }
+        try {
+            // try URL
+            return new URL(resourceLocation);
+        } catch (MalformedURLException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 144.